load-testing/tests/relays_stress_helpers_test.go

```diff
@@ -1034,148 +1033,195 @@ func (s *relaysSuite) sendRelay(iteration uint64, relayPayload string) (appAddre
 	gateway := s.activeGateways[iteration%uint64(len(s.activeGateways))]
 	application := s.activeApplications[iteration%uint64(len(s.activeApplications))]
 
-	gatewayUrl, err := url.Parse(s.gatewayUrls[gateway.address])
-	require.NoError(s, err)
-
-	// Include the application address in the query to the gateway.
-	query := gatewayUrl.Query()
-	query.Add("applicationAddr", application.address)
-	query.Add("relayCount", fmt.Sprintf("%d", iteration))
-	gatewayUrl.RawQuery = query.Encode()
-
-	// Use the pre-defined service ID that all application and suppliers are staking for.
-	gatewayUrl.Path = testedServiceId
-
 	// TODO_MAINNET: Capture the relay response to check for failing relays.
 	// Send the relay request within a goroutine to avoid blocking the test batches
 	// when suppliers or gateways are unresponsive.
-	go func(gwURL, payload string) {
-		_, err = http.DefaultClient.Post(
-			gwURL,
-			"application/json",
-			strings.NewReader(payload),
-		)
+	sendRelayRequest := func(gatewayURL, appAddr, payload string) {
+		gatewayURL = gatewayURL + "/"
+		req, err := http.NewRequest("POST", gatewayURL, strings.NewReader(payload))
+
+		// TODO_TECHDEBT(red-0ne): Use the app-address instead X-App-Address once PATH Gateway
+		// deprecates the X-App-Address header.
+		// This is needed by the PATH Gateway's trusted mode to identify the application
+		// that is sending the relay request.
+		req.Header.Add("X-App-Address", appAddr)
+		req.Header.Add("target-service-id", appAddr)
+		res, err := http.DefaultClient.Do(req)
 		require.NoError(s, err)
-	}(gatewayUrl.String(), relayPayload)
+
+		if res.StatusCode != http.StatusOK {
+			s.failedRelays.Add(1)
+		}
+	}
+
+	gatewayURL := s.gatewayUrls[gateway.address]
+	go sendRelayRequest(gatewayURL, application.address, relayPayload)
 
 	return application.address, gateway.address
 }
 
 // ensureFundedActors checks if the actors are funded by observing the transfer events
 // in the transactions results.
-func (s *relaysSuite) ensureFundedActors(
-	txResults []*types.TxResult,
-	actors []*accountInfo,
-) {
-	for _, actor := range actors {
-		actorFunded := false
-		for _, txResult := range txResults {
-			for _, event := range txResult.Result.Events {
-				// Skip non-relevant events.
-				if event.Type != "transfer" {
-					continue
-				}
-
-				attrs := event.Attributes
-				// Check if the actor is the recipient of the transfer event.
-				if actorFunded = hasEventAttr(attrs, "recipient", actor.address); actorFunded {
-					break
-				}
+func (s *relaysSuite) ensureFundedActors(ctx context.Context, actors []*accountInfo) {
+	if len(actors) == 0 {
+		return
+	}
+
+	fundedActors := make(map[string]struct{})
+	actorsAddrs := make([]string, len(actors))
+	for i, actor := range actors {
+		actorsAddrs[i] = actor.address
+	}
+
+	deadline := time.Now().Add(time.Second * time.Duration(blockDuration+1))
```

Comment:
NITs:

1. Should we rename `blockDuration` to `blockDurationSec`?
2. #PUC why +1